google/cloud/security/common/data_access/violation_dao.py

```diff
@@ -82,16 +84,21 @@ def insert_violations(self, violations,
         except MySQLdb.Error, e:
             raise db_errors.MySQLError(resource_name, e)
 
+        created_at = datetime.utcnow().strftime('%Y-%m-%d %H:%M:%S')
+
         inserted_rows = 0
         violation_errors = []
         for violation in violations:
             violation = self.Violation(
+                violation_hash=_create_violation_hash(
+                    violation['resource_id'], violation['violation_data']),
```

Comment:
Can resource_id and violation_data uniquely identify the violation? In 2.0 we are also using inventory_data.

Reply:
Yes, because there is some inventory data embedded in the violation_data.  For example:
https://github.com/GoogleCloudPlatform/forseti-security/blob/dev/google/cloud/security/scanner/scanners/cloudsql_rules_scanner.py#L63-L67